### PR TITLE
Fix hybrid AOT build

### PR DESCRIPTION
**Pull request by xmcclure**
Put #if WIN_PLATFORM around some WinRT timezone info functions

### CODE REVIEW

File: mcs/class/corlib/System/TimeZoneInfo.cs [hunk 1]
```diff
@@ -160,10 +160,13 @@ static TimeZoneInfo CreateLocal ()
 				name = TrimSpecial (name);
 				if (name != null)
 					return TimeZoneInfo.FindSystemTimeZoneById (name);
-			} else if (IsWindows) {
+			}
+#if WIN_PLATFORM
```
Comment (marek-safar) [inline mcs/class/corlib/System/TimeZoneInfo.cs]: Please fix the #if above to be `#if WIN_PLATFORM` only instead

File: mcs/class/corlib/System/TimeZoneInfo.cs [hunk 2]
```diff
@@ -215,11 +218,14 @@ static void GetSystemTimeZonesCore (List<TimeZoneInfo> systemTimeZones)
 				}
 
 				return;
-			} else if (IsWindows) {
+			}
+#if WIN_PLATFORM
```
Comment (marek-safar) [inline mcs/class/corlib/System/TimeZoneInfo.cs]: Please fix the #if above to be #if WIN_PLATFORM only instead

**Comment by xmcclure** [comment]
@marek-safar are you certain about those two changes? The enclosing #if blocks contain calls to things such as `FindSystemTimeZoneById`. I am not familiar with the history of this code but FindSystemTimeZoneById contains #if LIBC blocks and potentially loads time zone info from a file. It does not appear to be Windows specific code.

The enclosing is `#if (!FULL_AOT_DESKTOP || WIN_PLATFORM) && !XAMMAC_4_5`. Should the ENTIRE thing become WIN_PLATFORM? Or should it just be `#if WIN_PLATFORM && !XAMMAC_4_5`?

**Comment by xmcclure** [comment]
@marek-safar Is this what you meant? Or should the XAMMAC clause have been removed as well?

**Comment by marek-safar** [comment]
@xmcclure yes XAMMAC check is redundant

**Comment by xmcclure** [comment]
@monojenkins merge

**Pull request merged**